shotover-proxy/src/transforms/redis/timestamp_tagging.rs

```diff
@@ -122,38 +114,40 @@ fn unwrap_response(qr: &mut QueryResponse) {
                     }
                 }
             }
-            let mut hm = BTreeMap::new();
-            hm.insert("timestamp".to_string(), Value::List(timestamps));
 
-            qr.response_meta = Some(Value::Document(hm));
-            qr.result = Some(Value::List(results));
+            qr.response_meta = Some(Value::Document(BTreeMap::from([(
+                "timestamp".to_string(),
+                Value::List(timestamps),
+            )])));
+            Some(Value::List(results))
         } else if values.len() == 2 {
             qr.response_meta = values.pop().map(|v| {
-                let mut hm = BTreeMap::new();
-                if let Value::Integer(i, _) = v {
-                    let start = SystemTime::now();
-                    let since_the_epoch = start
-                        .duration_since(UNIX_EPOCH)
-                        .expect("Time went backwards");
-                    hm.insert(
-                        "timestamp".to_string(),
-                        Value::Integer(since_the_epoch.as_secs() as i64 - i, IntSize::I32),
-                    );
-                } else {
-                    hm.insert("timestamp".to_string(), v);
-                }
-                Value::Document(hm)
+                let processed_value = match v {
+                    Value::Integer(i, _) => {
+                        let start = SystemTime::now();
+                        let since_the_epoch = start
+                            .duration_since(UNIX_EPOCH)
+                            .expect("Time went backwards");
+                        Value::Integer(since_the_epoch.as_secs() as i64 - i, IntSize::I32)
```

Comment:
`as i64` with `IntSize::I32`?

Reply:
There is nothing inherently wrong with converting to i64 and setting the size to IntSize::I32 if it was originally i32. That was the design remember.

However looking at this, `as_secs` returns u64 so we should use IntSize::I64 here.